### PR TITLE
Remove dead code and fix warnings in the http2 module

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
Opportunities for clean up found while working on a different PR.

Modification:
* Dead code has been removed.
* Unnecessary parenthesis, qualifiers, etc. removed.
* Unused imports removed.
* Override annotations added where missing.

Result:
Cleaner code

### CODE REVIEW

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java [hunk 1]
```diff
@@ -681,7 +682,7 @@ public boolean isPushPromiseSent() {
          */
         private int nextReservationStreamId;
         private int lastStreamKnownByPeer = -1;
-        private boolean pushToAllowed = true;
+        private boolean pushToAllowed;
```
Comment (normanmaurer) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java:685]: This doesn't look correct as `false` is the default

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java [hunk 2]
```diff
@@ -555,7 +555,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
                               int padding, boolean endOfStream, ChannelPromise promise) {
             super(stream, padding, endOfStream, promise.unvoid());
             this.headers = headers;
-            this.hasPriorty = hasPriority;
+            hasPriorty = hasPriority;
```
Comment (normanmaurer) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java:558]: Better keep the `this` and just fix the typo ;)

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java (hunk 1 above)
Comment (chrisvest) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java:685]: The field is assigned in the constructor, so the initialiser is always overwritten.

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java:685]: ah ok... Then its fine :)

**Comment by chrisvest** [comment]
@normanmaurer Comments addressed

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
Ship it

**Pull request merged**